tp1/tp1.go

```diff
@@ -24,17 +48,93 @@ type Carrito struct {
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+	// check that my list is not empty
+	if len(ids) == 0 {
+		return nil
+	}
+
+	var carritos []Carrito
+	var supermercadoByPrice = make(map[string]int)
+
+	for _, id := range ids {
+		for _, productos := range p {
+			productoId, err := strconv.Atoi(productos[ID_INDEX])
```

Comment:
Podriamos utilizar un mapa de `ids` para evitarnos un for loop y tener acceso de O(1)

Reply:
hacer un mapa de Id por producto y rellenarlo al recorrer el for de productos?